src/transforms/sample.jl

```diff
@@ -18,50 +19,41 @@ Sample(1_000, replace=false, ordered=true)
 # with rng
 using Random
 rng = MersenneTwister(2)
-Sample(rng, 1_000)
+Sample(1_000, rng=rng)
 
 # with weights
-using StatsBase
-wv = pweights([0.2, 0.1, 0.3])
-Sample(wv, 1_000)
-Sample(rng, wv, 1_000)
+Sample(10, rand(100))
 ```
 """
-struct Sample{R<:AbstractRNG,W} <: Stateless
-  rng::R
-  wv::W
-  n::Int
+struct Sample{W,RNG} <: Stateless
+  size::Int
+  weights::W
   replace::Bool
   ordered::Bool
+  rng::RNG
 end
 
-Sample(rng::AbstractRNG, wv::AbstractWeights, n::Int; replace=true, ordered=false) =
-  Sample(rng, wv, n, replace, ordered)
-
-Sample(rng::AbstractRNG, n::Int; replace=true, ordered=false) =
-  Sample(rng, nothing, n, replace, ordered)
-
-Sample(wv::AbstractWeights, n::Int; replace=true, ordered=false) =
-  Sample(GLOBAL_RNG, wv, n, replace, ordered)
-
-Sample(n::Int; replace=true, ordered=false) =
-  Sample(GLOBAL_RNG, nothing, n, replace, ordered)
+Sample(size::Int, weights=[];
+       replace=false, ordered=false,
+       rng=Random.GLOBAL_RNG) =
+  Sample(size, weights, replace, ordered, rng)
 
 isrevertible(::Type{<:Sample}) = false
 
 function apply(transform::Sample, table)
   rows = Tables.rowtable(table)
 
-  rng     = transform.rng
-  wv      = transform.wv
-  n       = transform.n
+  size    = transform.size
+  weights = transform.weights
   replace = transform.replace
   ordered = transform.ordered
+  rng     = transform.rng
 
-  if isnothing(wv)
-    newrows = sample(rng, rows, n; replace, ordered)
+  newrows = if isempty(weights)
+    sample(rng, rows, size; replace, ordered)
   else
-    newrows = sample(rng, rows, wv, n; replace, ordered)
+    ws = Weights(collect(weights))
```

Comment:
I think this will limit the implementation, because StatsBase.jl defines other types of weights:
https://juliastats.org/StatsBase.jl/latest/weights/

Reply:
Yes, but I think we only ProbabilityWeights (or the generic Weights) in this use case. For end-users it is much more convenient to hide this internal statistical jargon and let them pass any vector of numbers as input, like one of the columns of the table.